### ISSUE TITLE
Autotype now includes window title per default

### ISSUE THREAD

**Issue by dploeger** [keeweb/keeweb#1814]
**Describe the bug**
I just upgraded from 1.17.6 to 1.18.3 and noticed, that Autotype includes filtering for the window title now and that breaks my workflow, because I have "Google" as a password in my database and now every password I want to autotype in Chrome only shows that google password (if I haven't added the specific website to the correct item).

I'm pretty sure, window titles were filtered before as well, but somehow this problem didn't show. I'm also unsure on how to fix this as the title can just as well be a good filter option. So in the solution I vouch for a configuration option which default filters to include for autotype. (Does that make it a FR instead?)

**To Reproduce**
1. Add a new password that includes "Google" in its name.
2. Open a website in Google Chrome
3. Hit autotype
4. Only see the Google password

**Expected behavior**
It could be configurable wether the window title is used as a default filter option for Autotype.

**Environment**
KeeWeb v1.18.3 (337ef44, 2021-05-09)
Environment: electron v12.0.7
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 11_3_1) AppleWebKit/537.36 (KHTML, like Gecko) KeeWeb/1.18.3 Chrome/89.0.4389.128 Electron/12.0.7 Safari/537.36

**Comment by antelle** [comment]
Interesting, I wonder how to fix this better. Probably we need a setting because I can imagine why someone would like to have the filter enabled by default.

It's also possible to remove standard browser names, such as Google Chrome, Firefox, etc... This can be another improvement.

**Comment by jivarson** [comment]
@dploeger, what's your stand on the below suggestions (coming from #1815)?

A. Hotkey in autotype-selection screen (Ctrl+A or such) to temporary disable all filters
B. Persistent setting to show / hide all filters
C. Persistent setting to show / hide specific filters (by default).

from my POV, the order is by preference, but I mainly want to get rid of the mouseclicks

**Comment by antelle** [comment]
There will be no preference to show / hide filters. However, we can add a preference to show filter as disabled by default.

**Comment by dploeger** [comment]
@jivarson I'm not sure if I really get those options. My opinion would be a preference to enable/disable certain filter types. That way I could just disable the window title filter for my specific problem.

Your idea about removing common browser names from the window title, @antelle, is a bit "hacky" in my opinion. That would be very hard to manage I think.

But I really have no really optimal solution here tbh.

**Comment by antelle** [comment]
I agree, it's possible that it's "hacky", I don't like this idea either, so probably it's better not to go there.

Looks like for now an option to disable the title filter would solve the problem.

**Comment by jivarson** [comment]
> @jivarson I'm not sure if I really get those options. My opinion would be a preference to enable/disable certain filter types. That way I could just disable the window title filter for my specific problem.
> 
> Your idea about removing common browser names from the window title, @antelle, is a bit "hacky" in my opinion. That would be very hard to manage I think.
> 
> But I really have no really optimal solution here tbh.

Realized I mixed the terms disable\hide above, sorry.
I think we're agreeing that a persistent setting to 'untick' the filter by default (maybe only the Title-filter as @antelle said) is the way to go..

**Comment by antelle** [comment]
Added the setting, let's see if it makes it easier, I'll build a release with it this week.

**Comment by antelle** [comment]
It's published: https://github.com/keeweb/keeweb/releases/latest

**Comment by dploeger** [comment]
Works like a charm. Thanks a lot!

**Comment by jivarson** [comment]
Perfectamento!

**Comment by kaibrune** [comment]
@antelle Sorry to ask but where can we find this option? By me it's still enabled by default and can't find a disable filter option. Thanks!

**Comment by antelle** [comment]
@kaibrune it's in Settings / General / Function. The filters won't disappear, but they won't be selected automatically.

**Comment by kaibrune** [comment]
> @kaibrune it's in Settings / General / Function. The filters won't disappear, but they won't be selected automatically.

![Screenshot](https://user-images.githubusercontent.com/33369042/132304221-0327b689-b821-44cb-8223-7d23800511c6.png)

Am I missing something? The "Filter entries by title in auto-type" is unchecked and still is the filter applied in the auto-type menu?

**Comment by antelle** [comment]
Which filter do you see?

**Comment by kaibrune** [comment]
> Which filter do you see?

![image](https://user-images.githubusercontent.com/33369042/132370985-90547d30-476f-49a2-9e98-aab65663e1e2.png)

I guess the window title filter?

**Comment by antelle** [comment]
@kaibrune I see, it happens because there's no URL. Then the setting is not applied, it should be fixed of course.

**Comment by kaibrune** [comment]
> @kaibrune I see, it happens because there's no URL. Then the setting is not applied, it should be fixed of course.

Ahh that's a point. For logins without a URL the filter is still applied. Would be awesome to disable it globally. :)

**Comment by jivarson** [comment]
> @kaibrune I see, it happens because there's no URL. Then the setting is not applied, it should be fixed of course.

This has been on my mind too, but I haven't taken the time to connect the dots. A fix would definitely speed up auto type on desktop apps.

**Comment by kaibrune** [comment]
Any updates on this matter? This is really blocking desktop use.

**Comment by XIDAGmbH** [comment]
I think this has still not been changed/fixed?
